karapace/schema_reader.py

```diff
@@ -50,7 +50,7 @@ class SchemaType(str, Enum):
 
 
 class TypedSchema:
-    def __init__(self, schema, schema_type: SchemaType, schema_str: str):
+    def __init__(self, schema: any, schema_type: SchemaType, schema_str: str):
```

Comment:
`schema` should be typed as `Union[avro.schema.Schema, jsonschema.Draft7Validator, dict]`. I'm not sure if the third option is necessary, we would need to investigate why we have this:

https://github.com/aiven/karapace/blob/ff348266f67517c644d8c4314c662599edb444a2/karapace/schema_reader.py#L307-L310

Reply:
And even type the `dict` in the `Union` should be typed.

Reply:
> And even type the dict in the Union should be typed

Maybe we should just remove it. Here is the PR that introduce the `json.loads`:

https://github.com/aiven/karapace/pull/91

I think we don't need it, just to handle the exception and log the message

Reply:
I removed all these typings I added. Out of scope of this PR.